docs/overview/off-chain-workers.md

```diff
@@ -0,0 +1,28 @@
+---
+title: "Substrate Off-chain Workers"
+---
+
+Many a times, there is a need of querying and/or processing of off-chain data before it can be included into the on-chain state. The conventional way of doing this is through oracles. Oracles are external services which, typically, listen to blockchain events and trigger tasks accordingly. When these tasks running inside the oracle complete, their result is submitted back to the blockchain using transactions. While this approach works, it still has several flaws w.r.t. security, scalability and infrastructure efficiency.
+
+To make the off-chain data integration secure and more efficient, Substrate includes off-chain workers. The off-chain worker subsystem allows execution of long-running tasks (web requests, encryption/decryption and signing of data, random number generation, etc.) which would require longer than the block execution time. 
+
+The off-chain workers have their own Wasm execution environment outside of the Substrate runtime. This separation of concerns is to make sure that the block production is not impacted by the long-running tasks. The communication between the off-chain workers and the Substrate runtime is done using `InherentData`. When a task in the off-chain worker completes execution, its result can be submitted back to the runtime using `InherentData`.
```

Comment:
> The communication between the off-chain workers and the Substrate runtime is done using `InherentData`. When a task in the off-chain worker completes execution, its result can be submitted back to the runtime using `InherentData`.

However as the off-chain workers are declared in the same code as the runtime, they can easily access on-chain state for their computations. To communicate results back on chain off-chain workers can submit extrinsics (signed and unsigned transactions) that are later included in subsequent blocks.